src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs

```diff
@@ -96,85 +109,81 @@ public async Task<VacancyEmployerInfoModel> GetVacancyEmployerInfoModelAsync(Vac
         }
 
         public async Task<OrchestratorResponse> PostLocationEditModelAsync(
-            LocationEditModel m, VacancyUser user, long ukprn, VacancyEmployerInfoModel employerInfoModel)
+            LocationEditModel locationEditModel, VacancyUser user, long ukprn,
+            VacancyEmployerInfoModel employerInfoModel)
         {
-            var vacancy = await Utility.GetAuthorisedVacancyForEditAsync(_providerVacancyClient, 
-                _recruitVacancyClient, m, RouteNames.Location_Post);
-            
-            var employerVacancyInfo = await _providerVacancyClient.GetProviderEmployerVacancyDataAsync(ukprn, vacancy.EmployerAccountId);
+            if (string.IsNullOrEmpty(locationEditModel.SelectedLocation))
+                return new OrchestratorResponse(false);
+
+            var vacancy = await Utility.GetAuthorisedVacancyForEditAsync(_providerVacancyClient,
+                _recruitVacancyClient, locationEditModel, RouteNames.Location_Post);
+
+            var employerVacancyInfo =
+                await _providerVacancyClient.GetProviderEmployerVacancyDataAsync(ukprn, vacancy.EmployerAccountId);
 
             LegalEntity selectedOrganisation;
```

Comment:
Can we just do what Employer.Web does regarding the setting of `legalEntityId` & `selectedOrganisation` ? Its confusing having two ways of doing the same thing.